src/layout.rs

```diff
@@ -292,11 +288,8 @@ impl Layout {
     ///     Rect::new(0, 8, 10, 2),
     /// ]);
     /// ```
-    pub fn constraints<C>(mut self, constraints: C) -> Layout
-    where
-        C: Into<Vec<Constraint>>,
-    {
-        self.constraints = constraints.into();
+    pub fn constraints<C: AsRef<[Constraint]>>(mut self, constraints: C) -> Layout {
```

Comment:
I'm curious. What is the advantage or difference of making this `AsRef`?

Reply:
Both `[T]` and `&[T]` are accepted

Reply:
The previous code had `Into<Vec<Constraint>>` and that seemed to accept `&[T]` as well, right? 

I guess my question is with this function being `AsRef<[Constaint]>` and the `Layout::new` being `Into<Vec<Constraint>>`, is there any difference from a user perspective?

Reply:
It was a mistake. This should be IntoIterator like the new() parameter

Reply:
I am going to go with `AsRef<[Constraint]>` here due to AsRef allowing slices of fixed sized arrays (which Into<Vec<Constraint>> does not. IntoIterator has an issue with another case.

```
   Compiling ratatui v0.23.0 (/Users/joshka/local/ratatui)
error[E0277]: the trait bound `Vec<layout::Constraint>: From<&[layout::Constraint; 2]>` is not satisfied
   --> src/layout.rs:634:43
    |
634 |             Layout::default().constraints(&CONSTRAINTS).constraints,
    |                               ----------- ^^^^^^^^^^^^ the trait `From<&[layout::Constraint; 2]>` is not implemented for `Vec<layout::Constraint>`
    |                               |
    |                               required by a bound introduced by this call
    |
    = note: required for `&[layout::Constraint; 2]` to implement `Into<Vec<layout::Constraint>>`
note: required by a bound in `layout::Layout::constraints`
   --> src/layout.rs:291:27
    |
291 |     pub fn constraints<C: Into<Vec<Constraint>>>(mut self, constraints: C) -> Layout {
    |                           ^^^^^^^^^^^^^^^^^^^^^ required by this bound in `Layout::constraints`
help: consider dereferencing here
    |
634 |             Layout::default().constraints(*&CONSTRAINTS).constraints,
    |                                           +

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ratatui` (lib test) due to previous error
warning: build failed, waiting for other jobs to finish...
```